python_on_whales/components/image/cli_wrapper.py

```diff
@@ -498,23 +502,26 @@ def pull(
         if x == []:
             return []
         elif isinstance(x, str):
-            return self._pull_single_tag(x, quiet=quiet)
+            return self._pull_single_tag(x, quiet=quiet, platform=platform)
         elif isinstance(x, list) and len(x) == 1:
-            return [self._pull_single_tag(x[0], quiet=quiet)]
+            return [self._pull_single_tag(x[0], quiet=quiet, platform=platform)]
         elif len(x) >= 2:
             pool = ThreadPool(4)
-            generator = self._generate_args_push_pull(x, quiet)
+            generator = self._generate_args_pull(x, quiet, platform)
             all_images = pool.starmap(self._pull_single_tag, generator)
             pool.close()
             pool.join()
             return all_images
 
-    def _pull_single_tag(self, image_name: str, quiet: bool):
+    def _pull_single_tag(self, image_name: str, quiet: bool, platform: str):
```

Comment:
Sorry I think these function signatures (also `_generate_args_pull()`) need to be changed to `platform: Optional[str]` since a `None` may be passed through now.